src/Backend/Modules/MediaLibrary/Domain/MediaItem/MediaItem.php

```diff
@@ -0,0 +1,603 @@
+     *
+     * @ORM\Column(type="datetime")
+     */
+    protected $editedOn;
+
+    /**
+     * @var ArrayCollection
+     *
+     * @ORM\OneToMany(
+     *     targetEntity="Backend\Modules\MediaLibrary\Domain\MediaGroupMediaItem\MediaGroupMediaItem",
+     *     mappedBy="item",
+     *     cascade={"persist","merge"},
+     *     orphanRemoval=true
+     * )
+     */
+    protected $groups;
+
+    /**
+     * MediaItem constructor.
+     *
+     * @param string $title
+     * @param string $url
+     * @param Type $type
+     * @param string $mime
+     * @param string $shardingFolderName
+     * @param MediaFolder $folder
+     * @param int $size
+     * @param int $userId
+     */
+    private function __construct(
+        string $title,
+        string $url,
+        Type $type,
+        string $mime,
+        string $shardingFolderName,
+        MediaFolder $folder,
+        int $size,
+        int $userId
+    ) {
+        $this->folder = $folder;
+        $this->userId = (int) $userId;
+        $this->type = $type;
+        $this->mime = $mime;
+        $this->shardingFolderName = $shardingFolderName;
+        $this->url = (string) $url;
+        $this->title = (string) $title;
+        $this->size = (int) $size;
+        $this->createdOn = new \DateTime();
+        $this->editedOn = new \DateTime();
+        $this->groups = new ArrayCollection();
+    }
+
+    /**
+     * @param string $source
+     * @param MediaFolder $folder
+     * @param int $userId
+     * @return MediaItem
+     * @throws \Exception
+     */
+    public static function createFromSource(
+        string $source,
+        MediaFolder $folder,
+        int $userId
+    ) {
+        try {
+            // Define file
+            $file = new File($source);
+
+            // We don't have a file
+            if (!$file->isFile()) {
+                throw new \Exception(
+                    'The given source is not a file.'
+                );
+            }
+
+            /** @var Type $mediaItemType */
+            $mediaItemType = Type::fromExtension($file->getExtension());
+
+            // Define sharding folder (getPath gets the path without the trailing slash)
+            /** @var string $shardingFolderName */
+            $shardingFolderName = basename($file->getPath());
+
+            // Define title
+            /** @var string $shardingFolderName */
+            $title = str_replace('.' . $file->getExtension(), '', $file->getFilename());
+
+            /** @var MediaItem $mediaItem */
+            $mediaItem = new self(
+                $title,
+                $file->getFilename(),
+                $mediaItemType,
+                $file->getMimeType(),
+                $shardingFolderName,
+                $folder,
+                $file->getSize(),
+                $userId
+            );
+
+            // Image
+            if ($mediaItemType->isImage()) {
+                try {
+                    list($width, $height) = getimagesize($source);
```

Comment:
I just tried your provided solution, but that gives me an error: "can't unpack ...".
So I would let this be...